### ISSUE TITLE
Look into replacing hashmaps with googles implementation.

### ISSUE THREAD

**Issue by billw2012** [caveman2cosmos/Caveman2Cosmos#282]
https://github.com/sparsehash/sparsehash
It isn't the absolute best performing, but it is C++03 compatible, and way better than any stl map implementations.

**Comment by OvermindDL1** [comment]
You might be interested in https://github.com/greg7mdp/sparsepp actually, it's based on Google's sparsehash but with a few game-centric performance and memory enhancements (along with a whole suite of benchmarks).  It's been since replaced with the same authors parallel-hashmap but that one requires at minimum C++11 instead.

**Comment by billw2012** [comment]
Thanks, I will try it out. I got a modest improvement for pathfinding cache with googles one already, hopefully this one can do better!

**Comment by OvermindDL1** [comment]
`sparsepp` fulfill's google's sparse_hash interface, so it's generally as simple as dropping it's source in and renaming a couple things if you are already using it (technically it's interface is even more simple as the tombstone types are no longer needed, but they just no-op in sparsepp anyway so the interface still stays the same).  :-)

**Comment by billw2012** [comment]
Yeah interface wise its definitely better, I can drop it in many places without needing to workout what tombstones to use
